### ISSUE TITLE
Is it possible to get a permanent fake value of SHA?

### ISSUE THREAD

**Issue by morting** [kkapsner/CanvasBlocker#64]
Hello. First of all, I want to thank you for the work done. It is very important for users who wish to remain anonymous.

Based on the data from the site http://www.browserleaks.com/canvas addon changes the parameter PNG Headers / IDAT and makes its digital random values. That leads to a whole new generation of SHA. If a mod Fake ...

I have a question. Is it possible to add an addon save the last used setting PNG Headers / IDAT parameter? Which will be stored until you close your browser, or to disable the settings.

Is it possible in the near future to get a fake SHA code that will not change when the page is refreshed? with the help of your addon.

Thank you!

**Comment by rugk** [comment]
Similar/Related to https://github.com/kkapsner/CanvasBlocker/issues/44

**Comment by kkapsner** [comment]
It's technically possible to create a cache BUT this would increase the memory usage of firefox (which is not too small without any addons) by a huge amount.

But that is the whole point of the fake mode to have different checksums every time. Why do you want to have a constant one?

**Comment by rugk** [comment]
Hmm... about the tags: I can understand the "wontfix", but why is the issue "invalid"?
Do you have a definiti0on? :wink:

**Comment by kkapsner** [comment]
Having a constant fingerprint increases the tracking ability. This defeats the purpose of the addon.

**Comment by kkapsner** [comment]
I still think a globally constant fingerprint is bad, but the ability to have the same "faking" for one session/tab for one domain can be a nice feature to have. I'm currently working on it.

**Comment by kkapsner** [comment]
The latest releatse 0.3.1 includes an option to switch to a persistent rng.

**Comment by rugk** [comment]
And it only works for one tab?

**Comment by kkapsner** [comment]
No - it is persistent per session (I'm a little bit surprised by that - I thought e10s would work differently).

**Comment by rugk** [comment]
> I'm a little bit surprised by that - I thought e10s would work differently

Ah, it depends on e10s. In this case e10s currently only separates tabs and the browser. Later one all tabs will be separated.
